doc/model.rst

```diff
@@ -0,0 +1,346 @@
+*****
+Model
+*****
+========
+Overview
+========
+Model(s) are subclasses of SmartSimEntity(s) and are created through the
+Experiment API. Models represent any computational kernel. Models are flexible
```

Comment:
"computational kernel" seems a little esoteric and maybe even ambiguous.  Can we make it more easily understood?  Also, I think this sentence might be better as the lead and maybe they don't need to know about ``SmartSimEntity`` because it's an abstract base class.  As a jumping off point for improvement, what about something like:

In the SmartSim API, ``Models`` are an abstract representation for compiled applications, scripts, and general computational tasks.  ``Models`` can be launched with other SmartSim entities and ML infrastructure to build AI-enabled workflows.  Models are flexible enough to support many different applications, however, to be used with SmartSim clients (SmartRedis) the application must be written in Python, C, C++, or Fortran.